scapy/layers/inet6.py

```diff
@@ -172,7 +175,7 @@ def parse_digit(a, netmask):
             netmask = min(8,max(netmask,0))
             a = (int(a) & (0xff<<netmask),(int(a) | (0xff>>(8-netmask)))+1)
             return a
-        self.parsed = map(lambda x,y: parse_digit(x,y), tmp, map(lambda x,nm=self.plen: x-nm, tuple))
+        self.parsed = list(map(lambda x,y: parse_digit(x,y), tmp, list(map(lambda x,nm=self.plen: x-nm, tuple))))
```

Comment:
The first list() is required to cast the map object to an actual list, but map allows iterators as arguments so the second one is not needed.